File: Java Basic/Testing System 6/src/com/vti/entity/Department.java
package com.vti.entity;

public class Department {
    public static int count;
    private int id;
    private String name;

    public Department(String name) {
        super();
        count++;
        this.id = count;
        this.name = name;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: ReactJS/Lesson/model/Department.ts
//export class
export class Department {
  private static counter: number = 0;

  private id: number;
  private name: string;

  constructor(name: string) {
    this.id = ++Department.counter;
    this.name = name;
  }

  public getId(): number {
    return this.id;
  }

  public getName(): string {
    return this.name;
  }

  /**
   * Setter id
   * @param {number} value
   */
  public setId(value: number) {
    this.id = value;
  }

  /**
   * Setter name
   * @param {string} value
   */
  public setName(value: string) {
    this.name = value;
  }

  public printDepartment() {
    console.log("**** Department: ****");
    console.log("ID: " + this.id);
    console.log("name: " + this.name);
  }
}
